src/install_host_app.sh

```diff
@@ -114,10 +114,10 @@ echo "Installing $BROWSER_NAME host config"
 # Create config dir if not existing
 mkdir -p "$TARGET_DIR"
 
-PATH_ESC="$(echo $PATH | sed -e 's/@/\\@/g')"
-PASS_PATH_ESC="$(echo $PASS_PATH | sed -e 's/@/\\@/g')"
-HOST_FILE_PATH_ESC="$(echo $HOST_FILE_PATH | sed -e 's/@/\\@/g')"
-PYTHON3_PATH_ESC="$(echo $PYTHON3_PATH | sed -e 's/@/\\@/g')"
+PATH_ESC="$(echo "$PATH" | sed -e 's/@/\\@/g')"
```

Comment:
Are you sure that this is correct syntax or rather: that it does what it's supposed to do?  Because you have double quotes inside of double quotes which looks like it will cause trouble...

Reply:
This is the recommendation of [SC2086](https://github.com/koalaman/shellcheck/wiki/SC2086). It is **really counter-intuitive**.

I'm wondering how $() is parsed internally...`( )` are used to create *subshells*...

> Note that $( ) starts a new context, and variables in it have to be quoted independently
> ```
> echo "This $variable is quoted $(but this $variable is not)"
> echo "This $variable is quoted $(and now this "$variable" is too)"
> ```

I'd like another source of information though.

**Otherwise**, we can just delete the enclosing quotes of `"$()"`, as illustrated below:
```
PATH_ESC=$(echo "$PATH" | sed -e 's/@/\\@/g')
```
Quotes are not necessary when assigning $(..) (but still good enough practice).

```
$ nasty_path="/usr/local/* /passff.py" # who does that???

# without enclosing quotes
$ result=$(echo "$nasty_path" | sed -e 's/@/\\@/g')
$ echo "$result" # do not forget the quotes! 
/usr/local/* /passff.py 

# with enclosing quotes
$ result="$(echo "$nasty_path" | sed -e 's/@/\\@/g')"
$ echo "$result" # do not forget the quotes! 
/usr/local/* /passff.py
```
:+1:

Reply:
Thanks for the explanation, very interesting.